engine/chain/internal/arc/service.go

```diff
@@ -1,23 +1,50 @@
 package arc
 
 import (
+	"context"
+
+	sdk "github.com/bitcoin-sv/go-sdk/transaction"
+	"github.com/bitcoin-sv/spv-wallet/engine/chain/errors"
+	"github.com/bitcoin-sv/spv-wallet/engine/chain/internal/ef"
 	"github.com/bitcoin-sv/spv-wallet/engine/chain/models"
 	"github.com/go-resty/resty/v2"
 	"github.com/rs/zerolog"
 )
 
 // Service for arc requests.
 type Service struct {
-	logger     zerolog.Logger
-	httpClient *resty.Client
-	arcCfg     chainmodels.ARCConfig
+	logger      zerolog.Logger
+	httpClient  *resty.Client
+	arcCfg      chainmodels.ARCConfig
+	efConverter interface {
+		Convert(ctx context.Context, tx *sdk.Transaction) (string, error)
+	}
 }
 
 // NewARCService creates a new arc service.
 func NewARCService(logger zerolog.Logger, httpClient *resty.Client, arcCfg chainmodels.ARCConfig) *Service {
-	return &Service{
+	service := &Service{
 		logger:     logger,
 		httpClient: httpClient,
 		arcCfg:     arcCfg,
 	}
+
+	if arcCfg.TxsGetter == nil {
+		logger.Warn().Msg("No transactions getter provided. Unsourced transactions will be broadcasted as raw hex.")
+		service.efConverter = &noTxsGettersEFConverter{}
+	} else {
+		service.efConverter = ef.NewConverter(arcCfg.TxsGetter)
+	}
+
+	return service
+}
+
+type noTxsGettersEFConverter struct{}
+
+func (n *noTxsGettersEFConverter) Convert(_ context.Context, tx *sdk.Transaction) (string, error) {
+	efHex, err := tx.EFHex()
+	if err != nil {
+		return "", chainerrors.ErrEFConversion.Wrap(err)
+	}
```

Comment:
you changed the logic and now it's not returning rawTx if EFHex will fail, or maybe I'm missing something 🤷

Reply:
You're missing something. 
Check that:
```golang
if arcCfg.TxsGetter == nil {
		logger.Warn().Msg("No transactions getter provided. Unsourced transactions will be broadcasted as raw hex.")
		service.efConverter = &noTxsGettersEFConverter{}
	} else {
		service.efConverter = ef.NewConverter(arcCfg.TxsGetter)
	}
```

```golang
func (s *Service) prepareTxHex(ctx context.Context, tx *sdk.Transaction) (string, error) {
	efHex, err := s.efConverter.Convert(ctx, tx)
	if errors.Is(err, context.DeadlineExceeded) || errors.Is(err, context.Canceled) {
		return "", spverrors.ErrCtxInterrupted.Wrap(err)
	}
	if err != nil {
		// Log level is set to Info because it can happen in standard flow when source transaction is not from our wallet (and Junglebus is disabled)
		s.logger.Info().Err(err).Msg("Could not convert transaction to EFHex. Using raw transaction hex as a fallback.")
		return tx.Hex(), nil
	}
	return efHex, nil
}
```

Reply:
ahhh ok, it was moved a little bit. 👍